### PR TITLE
Add public API baselines to SignalR

### PR DESCRIPTION
**Pull request by Pilchie**

### CODE REVIEW

**Comment by Pilchie** [comment]
Part of #24347

File: src/SignalR/server/Core/src/HubConnectionContext.cs [hunk 1]
```diff
@@ -146,6 +147,7 @@ internal StreamTracker StreamTracker
         // Currently used only for streaming methods
         internal ConcurrentDictionary<string, CancellationTokenSource> ActiveRequestCancellationSources { get; } = new ConcurrentDictionary<string, CancellationTokenSource>(StringComparer.Ordinal);
 
+        [SuppressMessage("ApiDesign", "RS0026:Do not add multiple overloads with optional parameters", Justification = "Required to maintain compatibility")]
```
Comment (dougbu) [inline src/SignalR/server/Core/src/HubConnectionContext.cs:150]: nit: Should we craft a better `Justification` for the many `T1 parameter, CancellationToken token = default`, `T2 parameter, CancellationToken token = default`, &hellip; cases❔ If we were adding `HubConnectionContext` tomorrow, I bet we'd use the same set of overloads. Perhaps,
```suggestion
        [SuppressMessage("ApiDesign", "RS0026:Do not add multiple overloads with optional parameters", Justification = "May need a CancellationToken with either of these message types.")]
```
Of course, if you do that here, we'd need to look at the rest of our open PRs and I'm not sure it's worthwhile.

**Review by dougbu — APPROVED** [review]

**Pull request merged**